platforms/software/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/PomReaderTest.groovy

```diff
@@ -160,6 +160,23 @@ class PomReaderTest extends AbstractPomReaderTest {
         pomReader.properties.containsKey('some.prop2')
     }
 
+    def "defaults to jar packaging when unable to replace properties"() {
+        when:
+        pomFile << """
+<project>
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>group</groupId>
+    <artifactId>artifact</artifactId>
+    <version>1.0</version>
+    <packaging>\${packaging.type}</packaging>
+</project>
+"""
+        pomReader = new PomReader(locallyAvailableExternalResource, moduleIdentifierFactory)
+
+        then:
+        pomReader.packaging == 'jar'
```

Comment:
💭 Related to the lack of a warning - wouldn't both this test (and the new integration test) pass prior to this change?  This doesn't have a way of verifying the changed behavior.